api/src/services/fields.ts

```diff
@@ -418,7 +418,12 @@ export class FieldsService {
 		if (field.schema?.has_auto_increment) {
 			column = table.increments(field.field);
 		} else if (field.schema?.data_type) {
-			column = table.specificType(field.field, field.schema.data_type);
+			// Append length to data type if it's stored in schema
+			const data_type =
+				field.schema?.max_length && field.schema.data_type !== 'longtext'
+					? `${field.schema.data_type}(${field.schema.max_length})`
```

Comment:
Is this the correct syntax cross-database?

Reply:
Yup 👍 It's always like this `datatype(length)`, although there are some types which doesn't accept the length property (as we have seen with `longtext`)
Also I tested it with MySQL, PostgreSQL and SQLite (as mentioned above)

Reply:
We should ideally rely on Knex to do that properly (eg `knex.text('field', maxLength)`) instead of re-adding that through here 🤔 

I believe I added that "specific data type" option at the top just to make sure `m2o` fields would use the exact same type as the primary key, in order to resolve some issues around `integer` vs `integer unsigned`, but the API handles that correctly now when setting up a relation, so we might be able to save the headache here by getting rid of this whole if statement 🤔

Reply:
Not sure if I can follow you, I mean we would still need to distinct between using `knex.text`, `knex.decimal`, ..., right?

Reply:
Which lines do you mean exactly with the "whole if statement"?

Reply:
Oh sorry for the ambiguity..

This specific type support:

https://github.com/directus/directus/blob/b5abb0fb20f1f8984ce506ea45727ddd8e5621e3/api/src/services/fields.ts#L420-L421

was added just as an escape hatch for the MySQL/MariaDB situation where "integer unsigned" is a dedicated type that doesn't exist elsewhere, but we might not need that anymore now that the relational setup is no longer used. Removing this if-statement completely should also solve the issue you are now trying to workaround 🙂

Reply:
Ah, I see! Thanks 😸 I can look at this tomorrow - it's getting late here 😴